bindings/profiler.cc

```diff
@@ -101,61 +105,226 @@ CpuProfiler* cpuProfiler = CpuProfiler::New(v8::Isolate::GetCurrent());
+      Nan::New<Integer>(node->GetScriptId()),
+      Nan::New<Integer>(node->GetLineNumber()),
+      Nan::New<Integer>(node->GetColumnNumber()),
+      Nan::New<Integer>(0),
+      children
+  );
+}
+#endif
 
 Local<Value> TranslateTimeProfileNode(const CpuProfileNode* node) {
-  Local<Object> js_node = Nan::New<Object>();
-  js_node->Set(Nan::New<String>("name").ToLocalChecked(),
-    node->GetFunctionName());
-  js_node->Set(Nan::New<String>("scriptName").ToLocalChecked(),
-    node->GetScriptResourceName());
-  js_node->Set(Nan::New<String>("scriptId").ToLocalChecked(),
-    Nan::New<Integer>(node->GetScriptId()));
-  js_node->Set(Nan::New<String>("lineNumber").ToLocalChecked(),
-    Nan::New<Integer>(node->GetLineNumber()));
-  js_node->Set(Nan::New<String>("columnNumber").ToLocalChecked(),
-    Nan::New<Integer>(node->GetColumnNumber()));
-  js_node->Set(Nan::New<String>("hitCount").ToLocalChecked(),
-    Nan::New<Integer>(node->GetHitCount()));
   int32_t count = node->GetChildrenCount();
   Local<Array> children = Nan::New<Array>(count);
   for (int32_t i = 0; i < count; i++) {
     children->Set(i, TranslateTimeProfileNode(node->GetChild(i)));
   }
-  js_node->Set(Nan::New<String>("children").ToLocalChecked(),
-    children);
-  return js_node;
+
+  return CreateTimeNode(
+    node->GetFunctionName(),
+    node->GetScriptResourceName(),
+    Nan::New<Integer>(node->GetScriptId()),
+    Nan::New<Integer>(node->GetLineNumber()),
+    Nan::New<Integer>(node->GetColumnNumber()),
+    Nan::New<Integer>(node->GetHitCount()),
+    children
+  );
 }
 
-Local<Value> TranslateTimeProfile(const CpuProfile* profile) {
+Local<Value> TranslateTimeProfile(const CpuProfile* profile, bool hasDetailedLines) {
   Local<Object> js_profile = Nan::New<Object>();
   js_profile->Set(Nan::New<String>("title").ToLocalChecked(),
     profile->GetTitle());
-  js_profile->Set(Nan::New<String>("topDownRoot").ToLocalChecked(),
+
+#if NODE_MODULE_VERSION > NODE_11_0_MODULE_VERSION
+  if (hasDetailedLines) {
+    js_profile->Set(
+      Nan::New<String>("topDownRoot").ToLocalChecked(),
+      TranslateLineNumbersTimeProfileRoot(profile->GetTopDownRoot()));
+  } else {
+    js_profile->Set(
+      Nan::New<String>("topDownRoot").ToLocalChecked(),
+      TranslateTimeProfileNode(profile->GetTopDownRoot()));
+  }
+#else
+  js_profile->Set(
+    Nan::New<String>("topDownRoot").ToLocalChecked(),
     TranslateTimeProfileNode(profile->GetTopDownRoot()));
+#endif
   js_profile->Set(Nan::New<String>("startTime").ToLocalChecked(),
     Nan::New<Number>(profile->GetStartTime()));
   js_profile->Set(Nan::New<String>("endTime").ToLocalChecked(),
     Nan::New<Number>(profile->GetEndTime()));
   return js_profile;
 }
 
+// Signature:
+// startProfiling(runName: string, includeLineInfo: boolean)
 NAN_METHOD(StartProfiling) {
-  Local<String> name = info[0].As<String>();
+  if (info.Length() != 2) {
+    return Nan::ThrowTypeError("StartProfling must have two arguments.");
```

Comment:
Do we consistently include the trailing dot in all error messages like this?  It's unusual, so making sure.

Reply:
The capitalization/punctuation of all thrown errors in profiler.cc is consistent, but maybe it should be changed.